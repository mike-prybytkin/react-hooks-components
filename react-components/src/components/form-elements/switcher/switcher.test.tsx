import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Switcher from './switcher';
import mockText from 'mocks/text';

const setUp = () =>
  render(
    <Switcher
      switcherRef={undefined}
      switcherType={mockText.switcherGender}
      optionOne={mockText.switcherGenderMale}
      optionTwo={mockText.switcherGenderFemale}
      idRadioOne={'one'}
      idRadioTwo={'two'}
    />
  );

describe('Switcher component', () => {
  it('should switch radio-button', () => {
    const { getByLabelText } = setUp();
    const radioMale = getByLabelText(mockText.switcherGenderMale) as HTMLInputElement;
    fireEvent.change(radioMale, { target: { value: mockText.switcherGenderFemale } });
    expect(radioMale.value).toBe(mockText.switcherGenderFemale);
  });

  it('should create snapshot', () => {
    const { asFragment } = setUp();
    expect(asFragment()).toMatchSnapshot();
  });
});
