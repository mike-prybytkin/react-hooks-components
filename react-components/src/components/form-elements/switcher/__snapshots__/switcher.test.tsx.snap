// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switcher component should create snapshot 1`] = `
<DocumentFragment>
  <div
    class="switcher-wrapper"
  >
    <p
      class="switcher-wrapper__header"
    >
      Gender:
    </p>
    <div
      class="switcher-wrapper__field"
    >
      <input
        checked=""
        id="one"
        name="switch-one"
        type="radio"
        value="Male"
      />
      <label
        for="one"
      >
        Male
      </label>
      <input
        id="two"
        name="switch-one"
        type="radio"
        value="Female"
      />
      <label
        for="two"
      >
        Female
      </label>
    </div>
  </div>
</DocumentFragment>
`;
